micro-rdk/src/esp32/board.rs

```diff
@@ -365,4 +380,44 @@ impl Board for EspBoard {
         }
         Err(BoardError::GpioPinError(pin as u32, "not configured"))
     }
+
+    fn add_digital_interrupt_callback(
+        &mut self,
+        pin: i32,
+        intr_type: InterruptType,
+        cb: gpio_isr_t,
+        arg: Option<*mut core::ffi::c_void>,
+    ) -> Result<(), BoardError> {
+        let p = self
+            .pins
+            .iter_mut()
+            .find(|p| p.pin() == pin)
+            .ok_or_else(|| {
+                BoardError::GpioPinError(
+                    pin as u32,
+                    "pin not found, failed to register interrupt callback",
+                )
+            })?;
+
+        if cb.is_some() {
+            p.setup_interrupt(intr_type, cb, arg.unwrap_or_else(core::ptr::null_mut))?;
+        } else {
+            let ptr = &mut p.event_count as *mut Arc<std::sync::atomic::AtomicU32> as *mut _;
+            p.setup_interrupt(
+                InterruptType::PosEdge,
+                Some(Esp32GPIOPin::default_interrupt),
+                ptr,
+            )?;
+        }
+
+        Ok(())
+    }
+
+    fn remove_digital_interrupt_callback(&mut self, pin: i32) -> Result<(), BoardError> {
+        unsafe {
+            esp!(gpio_isr_handler_remove(pin))
+                .map_err(|e| BoardError::GpioPinOtherError(pin as u32, Box::new(e)))?;
+        }
+        Ok(())
+    }
```

Comment:
will probably remove this. It was meant to be a wrapper for what the ultrasonic sensor does in its `impl Drop`, however the struct wouldn't have a handle to the board to use this method anyway.